File: practica1/src/baseDatos/Deserializador.java
package baseDatos;

import java.io.EOFException;
import java.io.File;
import java.io.FileInputStream;
import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.ObjectInputStream;
import java.util.ArrayList;
import java.util.List;

import gestorAplicacion.personal.Dependiente;
import gestorAplicacion.personal.Empleado;
import gestorAplicacion.personal.Tecnico;
import gestorAplicacion.tienda.Bodega;
import gestorAplicacion.tienda.CajaRegistradora;
import gestorAplicacion.tienda.Cliente;
import gestorAplicacion.tienda.Componente;
import gestorAplicacion.tienda.Producto;
import gestorAplicacion.tienda.Servicio;

/**
 * Clase para deserializar los objetos que se crearon en ejecucion
 * @author Erik Gonzalez
 * @author Felipe Miranda
 */
public class Deserializador {
	/**
	 * Utilizamos clases genericas para permitir reutilizar la funcion para todas
	 * las clases del proyecto
	 * 
	 * @param <E>       el generico se usa para poder agredar las clases que se
	 *                  crearon
	 * @param lista     Una lista de objetos
	 * @param className El nombre de la clase que queremos usar como nombre del
	 *                  archivo
	 */
	public static <E> void deserializador(List<E> list, String className) {
		FileInputStream fileIn;
		try {
			// Creamos una cadena con la ruta del archivo que vamos a cargar
			String path = System.getProperty("user.dir") + "/src/baseDatos/temp/" + className + ".txt";
			System.out.println(path);
			// utilizamos un file para crear este archivo si no existe aun
			File archivo = new File(path);
			archivo.createNewFile(); // Crea un nuevo archivo si no existe

			// usamos un FileInputStream para poder saber de donde cargar el archivo
			fileIn = new FileInputStream(path);

			// Si el archivo esta vacio se lanza un throw EOFException y se muestra como un
			// mensaje de vacio, pero si no se usa el objeto in para leer el archivo
			ObjectInputStream in = new ObjectInputStream(fileIn);

			// Lee el listado de elementos
			ArrayList<E> listado = (ArrayList<E>) in.readObject();

			// Recorro el ArrayList
			for (E el : listado) {
				list.add(el);
			}

			in.close();
			fileIn.close();

		} catch (FileNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		} catch (IOException e) {
			// TODO Auto-generated catch block
			System.out.println("Esta vacio");
		} catch (ClassNotFoundException e) {
			// TODO Auto-generated catch block
			e.printStackTrace();
		}
	}

	/**
	 * Funcion para deserializar toda la aplicacion Generic IT
	 */
	public static void deserializarTodo() {
		Deserializador.deserializador(Dependiente.getDependientes(), "Dependientes");
		Deserializador.deserializador(Tecnico.tecnicos, "Tecnicos");
		Deserializador.deserializador(CajaRegistradora.cajasRegistradoras, "CajasRegistradoras");
		Deserializador.deserializador(Cliente.getClientes(), "Clientes");
		Deserializador.deserializador(Componente.componentes, "Componentes");
		Deserializador.deserializador(Producto.productos, "Productos");
		Deserializador.deserializador(Servicio.getServicios(), "Servicios");
		Deserializador.deserializador(Bodega.getComponentes(), "Bodega");
		Deserializador.deserializador(Empleado.getEmpleados(), "Empleados");
	}
}

File: practica2/generic_it/base_datos/deserializador.py
import pickle
from gestor_aplicacion.personal.empleado import Empleado
from gestor_aplicacion.tienda.cliente import Cliente
from gestor_aplicacion.personal.tecnico import Tecnico
from gestor_aplicacion.personal.dependiente import Dependiente
from gestor_aplicacion.tienda.servicio import Servicio
from gestor_aplicacion.tienda.componente import Componente
from gestor_aplicacion.tienda.bodega import Bodega
from gestor_aplicacion.tienda.producto import Producto
from gestor_aplicacion.tienda.caja_registradora import CajaRegistradora
import pathlib
import os

# save_path = '/base_datos'
# file_name = "test.txt"

# completeName = os.path.join(save_path, file_name)
# print(completeName)
print(pathlib.Path(__file__).parent.absolute())

class Deserializador():
    
    def deserializar(lista, className):
        def camino(className):
            string = os.path.join(pathlib.Path(__file__).parent.absolute(), "temp\\"+className+".txt")
            return string
        # Leo el archivo
        try:
            picklefile = open(camino(className), 'rb')
        except:
            picklefile = open(camino(className), 'x')
            picklefile = open(camino(className), 'rb')
        # unpickle los datos
        if os.path.getsize(camino(className)) > 0:
            lista += pickle.load(picklefile)
        # Cierro el archivo
        picklefile.close()
    
    def deserializarTodo():
        Deserializador.deserializar(Dependiente.dependientes, "Dependientes")
        Deserializador.deserializar(Tecnico.tecnicos, "Tecnicos")
        Deserializador.deserializar(CajaRegistradora.cajasRegistradoras, "CajasRegistradoras")
        Deserializador.deserializar(Cliente.clientes, "Clientes")
        Deserializador.deserializar(Componente.componentes, "Componentes")
        Deserializador.deserializar(Producto.productos, "Productos")
        Deserializador.deserializar(Servicio.servicios, "Servicios")
        Deserializador.deserializar(Bodega._componentes, "Componentes")
        Deserializador.deserializar(Empleado._empleados, "Empleados")
        
    